server/eventlog.go

```diff
@@ -41,10 +76,27 @@ func ParseMachineState(rawEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState,
 		return &pb.MachineState{}, err
 	}
 
+	var grub *pb.GrubState
+	var kernel *pb.LinuxKernelState
+	if opts.Loader == GRUB {
+		grub, err = getGrubState(cryptoHash, rawEvents)
+		if err != nil {
+			// TODO(wuale): replace with SecureBoot changes for GroupedError
+			return &pb.MachineState{}, err
+		}
+		kernel, err = getLinuxKernelStateFromGRUB(grub, opts.ParseVerity)
```

Comment:
Instead of parsing both kernel cmdline and dmverity stuff in the same function, can they be done in two separate functions? I fell like it can make things more clear.

Reply:
like returning the parsed command line from its own function and then parsing dm-verity? That makes sense to me, will do.

Reply:
yes
```
kernel.commandline, err = getKernelCMDFromGRUB...
kernel.verity, err = getDmVerityFromGRUB...
```